src/hb/shape.rs

```diff
@@ -20,7 +20,16 @@ pub fn shape(face: &hb_font_t, features: &[Feature], mut buffer: UnicodeBuffer)
         buffer.0.language.as_ref(),
         features,
     );
-    shape_with_plan(face, &plan, buffer)
+    #[cfg(not(feature = "wasm-shaper"))]
```

Comment:
Shouldn't we check for the `Wasm` table here? I.e. this should be a runtime decision, not compile-time one.

If we have `Wasm` table, try `shape_with_wasm`, otherwise `shape_with_plan`.

Reply:
This is what this code does. The `Wasm` table is the first thing `shape_with_wasm` checks for, and if it is not there it returns `None`. The compile time check here is for the featuree as `shape_with_wasm` does not exist then.

Reply:
But if there is no `Wasm` table you exit the shaping immediately, instead of calling the regular shaper. No?
I would expect something like:

```python
if has_wasm_table and run_wasm_shaper():
  return

run_regular_shaper()
```

Reply:
yeah `shape_with_wasm` fails if there is no `"Wasm"` table. The fallback to regular shaper [happens in `shape_with_plan` here](https://github.com/asibahi/rustybuzz/blob/84b9f58d97d19a947064a0b0cc995c6ca7560bca/src/hb/shape.rs#L60). You can see the regular tests pass just fine with the feature turned on.